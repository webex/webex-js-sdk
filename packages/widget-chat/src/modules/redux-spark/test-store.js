// Store for testing
import {
  applyMiddleware,
  combineReducers,
  compose,
  createStore
} from 'redux';
import configureMockStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import {Map} from 'immutable';
import reducers from './reducers';

jest.mock(`./spark`);
const {createSpark} = require(`./spark`);

export function createMockStore() {
  const mockStore = configureMockStore([thunk]);
  return mockStore(new Map({
    status: new Map({
      authenticated: false,
      authenticating: false,
      registered: false,
      registering: false,
      connected: false,
      connecting: false
    }),
    spark: createSpark()
  }));
}

const enhancers = [
  applyMiddleware(
    thunk)
];

export const store = createStore(
  combineReducers({
    spark: reducers
  }),
  compose(...enhancers)
);
