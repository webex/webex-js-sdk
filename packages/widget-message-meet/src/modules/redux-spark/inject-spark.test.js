import React from 'react';
import {Provider} from 'react-redux';
import renderer from 'react-test-renderer';
import {store} from './test-store';

import TestComponent from '../../../test/test-component';

describe(`injectSpark`, () => {

  it(`renders correctly`, () => {
    const component = renderer.create(
      <Provider store={store}>
        <TestComponent />
      </Provider>
    ).toJSON();

    expect(component).toMatchSnapshot();
  });
});
