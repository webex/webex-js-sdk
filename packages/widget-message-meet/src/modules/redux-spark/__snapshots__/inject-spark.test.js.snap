exports[`injectSpark renders correctly 1`] = `
<div>
  <div>
    TestWidget
  </div>
</div>
`;
